sodasql/dialects/hive_dialect.py

```diff
@@ -0,0 +1,123 @@
+        self.configuration['hive.ddl.output.format']='json'
+        try:
+            conn = hive.connect(
+                username=self.username,
+                password=self.password,
+                host=self.host,
+                port=self.port,
+                database=self.database,
+                configuration={ key: str(value) for key, value in self.configuration.items() }, #https://github.com/jaegertracing/jaeger-client-python/issues/151
+                auth=None)
+            return conn
+        except Exception as e:
+            self.try_to_raise_soda_sql_exception(e)
+
+    def sql_columns_metadata(self, table_name:str):
+        # getting columns info from hive which version <3.x needs to be parsed
+        column_tuples = []
+        cur = self.create_connection().cursor()
+        cur.execute(f"describe {self.database}.{table_name}")
+        result = cur.fetchall()[0][0]
+        if result is not None:
+            result_json = json.loads(result)
+            for column in result_json['columns']:
+                column_tuples.append((column['name'],column['type'], 'YES'))
+        return column_tuples
+
+    def sql_columns_metadata_query(self, table_name: str) -> str:
+        # hive_version < 3.x does not support information_schema.columns
+        return ''
+
+    def qualify_table_name(self, table_name: str) -> str:
+        return f'{self.database}.{table_name}'
+
+    def qualify_writable_table_name(self, table_name: str) -> str:
+        return self.qualify_table_name(table_name)
+
+    def sql_expr_regexp_like(self, expr: str, pattern: str):
+        return f"{expr} rlike '{self.qualify_regex(pattern)}'"
+
+    def is_text(self, column_type: str):
+        for text_type in self._get_text_types():
+            if column_type and text_type.upper() in column_type.upper():
+                return True
+        return False
+
+    def _get_text_types(self):
+        return ['CHAR', 'VARCHAR', 'STRING']
+
+    def is_number(self, column_type: str):
+        for number_type in self._get_number_types():
+            if column_type and number_type.upper() in column_type.upper():
+                return True
+        return False
+
+    def _get_number_types(self):
+        return ['TINYINT', 'SMALLINT', 'INT', 'BIGINT', 'FLOAT', 'DOUBLE', 'DOUBLE PRECISION', 'DECIMAL', 'NUMERIC']
+
+    def sql_expr_stddev(self, expr: str):
+        return f'STDDEV_POP({expr})'
+
+    def qualify_regex(self, regex) -> str:
+        return self.escape_metacharacters(regex).replace("'", "\\'")
+
+    def is_connection_error(self, exception):
+        if exception is None:
+            return False
+        return isinstance(exception, Error)
+
+    def is_authentication_error(self, exception):
+        if exception is None:
+            return False
+        return isinstance(exception, TTransportException)
+
+    def try_to_raise_soda_sql_exception(self, exception: Exception) -> Exception:
```

Comment:
This override is probably not necessary - right?

Reply:
yes. I removed it from hive_dialect.py